src/sinks/datadog/metrics.rs

```diff
@@ -259,7 +272,7 @@ fn build_uri(host: &str, endpoint: &'static str) -> crate::Result<Uri> {
 }
 
 async fn healthcheck(config: DatadogConfig, client: HttpClient) -> crate::Result<()> {
-    let uri = format!("{}/api/v1/validate", config.get_endpoint())
+    let uri = format!("{}/api/v1/validate", config.get_api_endpoint())
```

Comment:
Is there no healthcheck route we could use on `https://{}-vector.agent.datadoghq.com` (and friends)? I think checking that endpoint would be preferable as it gives a higher level of confidence than just checking `api.`.

Reply:
When sending logs to `https://api.<site>` as we were doing before this PR, they ultimately end in the `*.agent.<SITE>` pipeline, that we now directly hit through `https://{}-vector.agent.<site>` endpoints and they do not exactly support the `/api/v1/validate` but they rather reply with a http redirect to `https://api.<site>`. I kept the current agent core logic that actually relies on `https://api.<site>` for health-checking as this is how the agent is doing its health-check (https://github.com/DataDog/datadog-agent/blob/master/pkg/forwarder/forwarder_health.go#L131-L143).

Reply:
Aha, gotcha, thanks for clarifying. 👍 to matching `datadog-agent`'s behavior.

Reply:
I added a comment to reflect that as the reason for doing that might be confusing.